src/vs-bicep/Bicep.VSLanguageServerClient/ContentType/BicepConfigContentTypeProvider.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Bicep.VSLanguageServerClient.ContentType
+{
+    /// <summary>
+    /// VS lsp sends out events to languages based on content type and file extension.
+    /// Content types are listed in <see cref="BicepLanguageServerClient"/>. Since we want to listen to changes in 
+    /// bicepconfig.json files, we will create a separate content type and list it in <see cref="BicepLanguageServerClient"/>.
+    /// We specifically set the fileName here to make sure events get sent across only for bicepconfig.json changes.
+    /// </summary>
+    [Export(typeof(IFilePathToContentTypeProvider))]
+    [Name("BicepConfigContentTypeProvider")]
+    [FileName("bicepconfig.json")]
+    public class BicepConfigContentTypeProvider : IFilePathToContentTypeProvider
+    {
+        private readonly IContentTypeRegistryService contentTypeRegistryService;
+
+        [ImportingConstructor]
+        public BicepConfigContentTypeProvider(IContentTypeRegistryService contentTypeRegistryService)
+        {
+            this.contentTypeRegistryService = contentTypeRegistryService;
+        }
+
+        public bool TryGetContentTypeForFilePath(string filePath, out IContentType? contentType)
+        {
+            var fileName = Path.GetFileName(filePath);
+
+            if (fileName.Equals("bicepconfig.json", StringComparison.Ordinal))
```

Comment:
bicep in vscode seems to ignore casing of bicepconfig.json filename

Reply:
Made the check case insensitive:
https://github.com/Azure/bicep/commit/9fec68204f2c6fae147424d4b7552f2ec16bfea9

Thanks!